src/libmr_commands.c

```diff
@@ -11,6 +11,18 @@
 
 #include "rmutil/alloc.h"
 
+static inline bool check_and_reply_on_error(ExecutionCtx *eCtx, RedisModuleCtx *rctx) {
+    size_t len = MR_ExecutionCtxGetErrorsLen(eCtx);
+    bool has_err = false;
+    for (int i = 0; i < len; i++) {
+        RedisModule_ReplyWithError(rctx, MR_ExecutionCtxGetError(eCtx, i));
```

Comment:
Either return an array (RM_ReplyWithArray ...) or just return the first error ...